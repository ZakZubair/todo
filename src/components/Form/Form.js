import React from 'react';
import PropTypes from 'prop-types';
import Input from '../Input';
import Button from '../Button';

const Form = ({
  name, description, onChangeName, onChangeDescription, onCreate, onKeyPress,
}) => (
  <div className="form">
    <Input name="name" value={name} onChange={onChangeName} placeholder="Name" />
    <Input name="description" value={description} onChange={onChangeDescription} placeholder="Description" onKeyPress={onKeyPress} />
    <Button onClick={onCreate}>+</Button>
  </div>
);

Form.propTypes = {
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  onChangeName: PropTypes.func.isRequired,
  onChangeDescription: PropTypes.func.isRequired,
  onCreate: PropTypes.func.isRequired,
  onKeyPress: PropTypes.func.isRequired,
};
export default Form;
