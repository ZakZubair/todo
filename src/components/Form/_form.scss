.form {
  display: flex;
  flex-direction: column;
  background-color: $white;
  padding: rem(16) rem(16) rem(24);
}
