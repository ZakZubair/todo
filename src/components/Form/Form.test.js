import 'jsdom-global/register';
import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Form from './Form';

describe('<Form />', () => {
  it('renders name and description with add button', () => {
    const mockCallBack = jest.fn();

    const props = {
      name: 'Test name',
      description: 'Test description',
      onChangeName: mockCallBack,
      onChangeDescription: mockCallBack,
      onCreate: mockCallBack,
      onKeyPress: mockCallBack,
    };

    const wrapper = mount(<Form {...props} />);
    const name = wrapper.find('input').at(0);
    const description = wrapper.find('input').at(1);
    const button = wrapper.find('button');
    expect(name).toHaveLength(1);
    name.simulate('change', { target: { value: 'New title' } });
    expect(description).toHaveLength(1);
    description.simulate('change', { target: { value: 'New description' } });
    expect(button).toHaveLength(1);
    expect(button.prop('type')).toEqual('button');
    expect(button.text()).toEqual('+');
  });

  it('render correctly form component', () => {
    const mockCallBack = jest.fn();

    const props = {
      name: 'Test name',
      description: 'Test description',
      onChangeName: mockCallBack,
      onChangeDescription: mockCallBack,
      onCreate: mockCallBack,
      onKeyPress: mockCallBack,
    };

    const FormComponent = renderer.create(<Form {...props} />).toJSON();
    expect(FormComponent).toMatchSnapshot();
  });
});
