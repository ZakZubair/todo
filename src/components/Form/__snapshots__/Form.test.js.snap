// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> render correctly form component 1`] = `
<div
  className="form"
>
  <input
    className="input"
    onChange={[MockFunction]}
    onClick={null}
    onKeyPress={null}
    placeholder="Name"
    value="Test name"
  />
  <input
    className="input"
    onChange={[MockFunction]}
    onClick={null}
    onKeyPress={[MockFunction]}
    placeholder="Description"
    value="Test description"
  />
  <button
    className="button__add"
    onClick={[MockFunction]}
    type="button"
  >
    +
  </button>
</div>
`;
